src/phpDocumentor/ConfigurationFactory/PhpDocumentor3.php

```diff
@@ -0,0 +1,197 @@
+ */
+final class PhpDocumentor3 implements Strategy
+{
+    /**
+     * The path to the xsd that is used for validation of the configuration file.
+     *
+     * @var string
+     */
+    private $schemaPath;
+
+    /**
+     * Initializes the PhpDocumentor3 strategy.
+     *
+     * @param string $schemaPath
+     */
+    public function __construct($schemaPath)
+    {
+        if ($schemaPath === '') {
+            $schemaPath = __DIR__ . '../../../../data/xsd/phpdoc.xsd';
+        }
+
+        $this->schemaPath = $schemaPath;
+    }
+
+    /**å
+     * Converts the phpDocumentor3 configuration xml to an array.
+     *
+     * @param \SimpleXMLElement $phpDocumentor
+     *
+     * @return array
+     */
+    public function convert(\SimpleXMLElement $phpDocumentor)
+    {
+        $this->validate($phpDocumentor);
+
+        $versions = [];
+        $template = [];
+
+        foreach ($phpDocumentor->children() as $key => $value) {
+            switch ((string) $key) {
+                case 'version':
+                    $versions[(string) $value->attributes()->number] = $this->buildVersions($value);
+                    break;
+                case 'template':
+                    $template = $this->buildTemplate($value);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        $outputDirectory = ((string) $phpDocumentor->parser->target) ?: 'file://build/docs';
+        $cacheDirectory  = ((string) $phpDocumentor->parser->cache) ?: '/tmp/phpdoc-doc-cache';
+
+        $phpdoc3Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => $outputDirectory,
+                    'cache'  => $cacheDirectory,
+                ],
+                'versions'  => $versions,
+                'templates' => $template,
+            ],
+        ];
+
+        return $phpdoc3Array;
+    }
+
+    /**
+     * Checks if the provided configuration xml is for phpDocumentor3.
+     *
+     * @param $phpDocumentor
+     *
+     * @return bool
+     */
+    public function match(\SimpleXMLElement $phpDocumentor)
+    {
+        return (int) $phpDocumentor->attributes()->version === 3;
+    }
+
+    /**
+     * Builds the versions part of the array from the phpDocumentor3 configuration xml.
+     *
+     * @param \SimpleXMLElement $version
+     *
+     * @return array
+     */
+    private function buildVersions(\SimpleXMLElement $version)
+    {
+        $extensions = [];
+        foreach ($version->api->extensions->children() as $extension) {
+            if ((string) $extension !== '') {
+                $extensions[] = (string) $extension;
+            }
+        }
+
+        $ignoreHidden   = filter_var($version->api->ignore->attributes()->hidden, FILTER_VALIDATE_BOOLEAN);
+        $ignoreSymlinks = filter_var($version->api->ignore->attributes()->symlinks, FILTER_VALIDATE_BOOLEAN);
```

Comment:
ignore-symlinks may be removed from the entire phpDocumentor 3 spec; with the introduction of FlySystem we won't be supporting symlinks anymore. (you couldn't have known this; this came up during implementation of FlySystem)